### ISSUE TITLE
`unary_union` producing slightly different results

### ISSUE THREAD

**Issue by freemvmt** [Toblerity/Shapely#1193]
## Expected behavior and actual behavior.

I am running `ops.unary_union` on a big list of polygons in order to 'dissolve' them together, but getting inconsistent outputs.

I am running this same operation locally on my macOS Catalina 10.15.7, and on an Ubuntu 18.04.5 LTS docker instance on circleCI. In both cases, I'm getting a `MultiPolygon` that on the face of it is exactly the same shape, except that the polygons therein are ordered ever so slightly differently - two of them are swapped.

I am wondering whether the output of `unary_union` is fully deterministic given the same input?

In both cases I'm running the same Python, Shapely and GEOS versions, as far as I can tell.

## Steps to reproduce the problem.

Here's the list of polygons as an input WKT `MultiPolygon`: https://gist.github.com/freemvmt/15a203a62121440e11df3d1744cbb281

To do the same operation, make sure you're using the same python/shapely/geos versions, and run the following:

```python
from shapely import ops, wkt

multi_polygon = wkt.loads(input)
shape = ops.unary_union(multi_polygon)
print(wkt.dumps(shape))
```

My output from this locally (on the mac) is: https://gist.github.com/freemvmt/84d683534f9c1d3fd0ce1d9ff1cdf7e5
However, my output on the Linux instance is: https://gist.github.com/freemvmt/c7a000fd98c11a3dd515feac33e2dba8

It's the following part that is at issue - in the first case:

```
..., ((-0.1246099999999986 51.5190899999999985, -0.1245800000000000 51.5191099999999977, -0.1245500000000000 51.5190800000000024, -0.1245900000000000 51.5190699999999993, -0.1246099999999986 51.5190899999999985)), ((-0.1154100000000000 51.5190899999999985, -0.1154000000000000 51.5190699999999993, -0.1153300000000000 51.5190800000000024, -0.1153400000000000 51.5191099999999977, -0.1154100000000000 51.5191000000000017, -0.1154100000000000 51.5190899999999985)), ...
```

and in the second case, these lists of coords are swapped:

```
..., ((-0.1154100000000000 51.5190899999999985, -0.1154000000000000 51.5190699999999993, -0.1153300000000000 51.5190800000000024, -0.1153400000000000 51.5191099999999977, -0.1154100000000000 51.5191000000000017, -0.1154100000000000 51.5190899999999985)), ((-0.1246099999999986 51.5190899999999985, -0.1245800000000000 51.5191099999999977, -0.1245500000000000 51.5190800000000024, -0.1245900000000000 51.5190699999999993, -0.1246099999999986 51.5190899999999985)), ...
```

## Operating system

macOS Catalina 10.15.7 vs Ubuntu 18.04.5 LTS 

## Shapely version and provenance

- Python version: `3.6.7`
- Shapely version: `1.7.1`

As for GEOS, I _think_ in both cases I am using the version bundled with Shapely, which I _think_ is `GEOS 3.6.0-CAPI-1.10.0` (reading this from `shapely/geos.py`). I deliberately uninstalled my homebrew GEOS to try and ensure this was the case.

I could be wrong about the GEOS versions being the same, in which case it could be what is at issue, as per [this old issue](https://github.com/Toblerity/Shapely/issues/816). Does anyone know how to determine exactly what GEOS is being used when `unary_union` runs? In particular, whether it's the bundled one, or another installation elsewhere on the OS?

**Comment by mwtoews** [comment]
Good to see that all results provide valid and equal geometries. The fix that will provide consistent ordering of geometry information is `shape.normalize()` (#1090), which is a feature that is part of (soon to be released) Shapely 1.8.

With valid inputs, different GEOS versions should deterministically return "equal" geometries, but the internal ordering may differ to give different (e.g.) WKT. If/when you can, use the `.normalize()` method to clean-up the geometries before comparing their serialized outputs.

Also, you can get the GEOS version used by shapely with:
```python
from shapely import geos
print(geos.geos_version_string)
```

**Comment by freemvmt** [comment]
So, the `geos.geos_version_string` pointer helped me conclude that the same GEOS version was in use.

Since Shapely 1.8 is still in alpha, I've written a custom solution for the normalization in the meantime. This has solved the issue for me so I'll close it.

Thanks so much for your help @mwtoews! 🙌🏼
